crates/sequencer_proof_client/src/lib.rs

```diff
@@ -94,4 +102,14 @@ pub trait ProofClient {
         to_block_number: L2BlockNumber,
         proof: SnarkWrapperProof,
     ) -> anyhow::Result<()>;
+    async fn peek_fri_job(&self, block_number: u32) -> anyhow::Result<Option<(u32, Vec<u8>)>>;
```

Comment:
Do we want this set of functions in the trait? Maybe not. For instance, it makes no sense to do anything for file based different during peak/pick. I'd much rather either add a new trait for this (`PeekableProofClient`?) and implement it only where needed (sequencer).

WDYT?